test/unit/search/fm_index/fm_index_dna4_test.cpp

```diff
@@ -5,21 +5,46 @@
 // shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
 // -----------------------------------------------------------------------------------------------------
 
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-
-#include <seqan3/alphabet/nucleotide/dna4.hpp>
 #include "fm_index_collection_test_template.hpp"
 #include "fm_index_test_template.hpp"
 
-using t1 = std::pair<fm_index<text_layout::single>, std::vector<dna4>>;
+using t1 = std::pair<fm_index<dna4, text_layout::single>, std::vector<dna4>>;
 INSTANTIATE_TYPED_TEST_CASE_P(dna4, fm_index_test, t1);
-using t2 = std::pair<fm_index<text_layout::collection>, std::vector<std::vector<dna4>>>;
+using t2 = std::pair<fm_index<dna4, text_layout::collection>, std::vector<std::vector<dna4>>>;
 INSTANTIATE_TYPED_TEST_CASE_P(dna4_collection, fm_index_collection_test, t2);
 
-using t3 = std::pair<fm_index<false>, std::vector<dna4>>;
-INSTANTIATE_TYPED_TEST_CASE_P(dna4_deprecation, fm_index_test, t3);
-
 TEST(fm_index_test, additional_concepts)
 {
     EXPECT_TRUE(detail::SdslIndex<default_sdsl_index_type>);
 }
+
+TEST(fm_index_test, cerealisation_errors)
+{
+#if SEQAN3_WITH_CEREAL
+    using namespace seqan3;
+
+    fm_index<dna4, text_layout::single> index{"AGTCTGATGCTGCTAC"_dna4};
+
+    test::tmp_filename filename{"cereal_test"};
+
+    {
+        std::ofstream os{filename.get_path(), std::ios::binary};
+        cereal::BinaryOutputArchive oarchive{os};
+        oarchive(index);
+    }
+
```

Comment:
reading it in again is missing.

Reply:
The following two scopes are read in. 
This test is for reading into incorrect types (Should throw).
The test_template contains the test for correct reading.

Reply:
ah thank you